sql: panic running SHOW CLUSTER SESSIONS during partition

On @jseldess's geographically distributed 3 node Digital Ocean cluster, I started a `SHOW CLUSTER SESSIONS` query while 2 of the 3 nodes were down. The query hung, for tens of seconds, then I started the other two nodes back up. They started up successfully, but the node that the query was running on panic'ed:

```
W170922 18:45:41.286322 72788 sql/crdb_internal.go:805  [client=omitted,user=root,n1] failed to get sessions from node 3: rpc error: code = Unavailable desc = grpc: the connection is unavailable
E170922 18:45:41.286400 72788 sql/session.go:1335  [client=omitted,user=root,n1] a SQL panic has occurred!
E170922 18:45:41.286648 72788 util/log/crash_reporting.go:114  [n1] a panic has occurred!
panic: datum row count and column count differ: 8 vs 9 [recovered]
	panic: panic while executing "show cluster sessions;": datum row count and column count differ: 8 vs 9 [recovered]
	panic: panic while executing "show cluster sessions;": datum row count and column count differ: 8 vs 9 [recovered]
	panic: panic while executing "show cluster sessions;": datum row count and column count differ: 8 vs 9

goroutine 72788 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc4200915f0, 0x7fb65ac915f8, 0xc420124fc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:183 +0xb2
panic(0x1b3b340, 0xc423f98fe0)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).serve.func1(0xc421e3d200, 0x7fb65ac915f8, 0xc420c3aba0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:423 +0xa1
panic(0x1b3b340, 0xc423f98fe0)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/cockroachdb/cockroach/pkg/sql.(*Session).maybeRecover(0xc422493000, 0x1cfe948, 0x9, 0xc421aeb4b3, 0x16)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/session.go:1346 +0x28d
panic(0x1a7fc40, 0xc42037a610)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/cockroachdb/cockroach/pkg/sql.virtualTableEntry.getPlanInfo.func1.1(0xc4230c0d80, 0x8, 0x8, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_schema.go:123 +0x573
github.com/cockroachdb/cockroach/pkg/sql.populateSessionsTable(0x7fb65ac918b8, 0xc424731f80, 0xc4203bcb40, 0xc420d407e0, 0xc420d407e0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/crdb_internal.go:817 +0xa8e
github.com/cockroachdb/cockroach/pkg/sql.glob..func14(0x7fb65ac918b8, 0xc424731f80, 0xc422493208, 0x0, 0x0, 0xc4203bcb40, 0xc4203bca20, 0xc422493938)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/crdb_internal.go:754 +0xf2
github.com/cockroachdb/cockroach/pkg/sql.virtualTableEntry.getPlanInfo.func1(0x7fb65ac918b8, 0xc424731f80, 0xc422493208, 0x0, 0x0, 0x0, 0x3000000000009, 0xc, 0x1b)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_schema.go:134 +0x110
github.com/cockroachdb/cockroach/pkg/sql.(*planner).getVirtualDataSource.func1(0x7fb65ac918b8, 0xc424731f80, 0xc422493208, 0x5be6e2, 0xc421798000, 0x280, 0x280)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/data_source.go:319 +0x5c
github.com/cockroachdb/cockroach/pkg/sql.doExpandPlan(0x7fb65ac918b8, 0xc424731f80, 0xc422493208, 0x7fffffffffffffff, 0x0, 0x0, 0x0, 0x2aba780, 0xc421356140, 0x10, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/expand_plan.go:221 +0x658
github.com/cockroachdb/cockroach/pkg/sql.expandRenderNode(0x7fb65ac918b8, 0xc424731f80, 0xc422493208, 0x7fffffffffffffff, 0x0, 0x0, 0x0, 0xc4244a3200, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/expand_plan.go:310 +0xda
github.com/cockroachdb/cockroach/pkg/sql.doExpandPlan(0x7fb65ac918b8, 0xc424731f80, 0xc422493208, 0x7fffffffffffffff, 0x0, 0x0, 0x0, 0x2abac00, 0xc4244a3200, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/expand_plan.go:217 +0x167e
github.com/cockroachdb/cockroach/pkg/sql.(*planner).expandPlan(0xc422493208, 0x7fb65ac918b8, 0xc424731f80, 0x2abac00, 0xc4244a3200, 0x2abac00, 0xc4244a3200, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/expand_plan.go:33 +0x93
github.com/cockroachdb/cockroach/pkg/sql.(*planner).optimizePlan(0xc422493208, 0x7fb65ac918b8, 0xc424731f80, 0x2abac00, 0xc4244a3200, 0xc421e4a140, 0x9, 0x9, 0x2abac00, 0xc4244a3200, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/optimize.go:46 +0x10b
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makePlan(0xc422493208, 0x7fb65ac918b8, 0xc424731f80, 0x7fb6568efff8, 0xc423f83c78, 0x0, 0x0, 0x0, 0x14e6c341ef0e7e7b, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:202 +0x166
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmt(0xc420408480, 0x7fb6568efff8, 0xc423f83c78, 0x0, 0x0, 0x0, 0x14e6c341ef0e7e7b, 0x1, 0xc4210612c0, 0xc422493208, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1886 +0x104
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtInOpenTxn(0xc420408480, 0xc422493000, 0x7fb6568efff8, 0xc423f83c78, 0x0, 0x0, 0x0, 0x14e6c341ef0e7e7b, 0x1, 0xc4210612c0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1601 +0x6e3
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execSingleStatement(0xc420408480, 0xc422493000, 0x7fb6568efff8, 0xc423f83c78, 0x0, 0x0, 0x0, 0x14e6c341ef0e7e7b, 0x1, 0xc4210612c0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1174 +0x92b
github.com/cockroachdb/cockroach/pkg/sql.runTxnAttempt(0xc420408480, 0xc422493000, 0xc424731f00, 0x1, 0x1, 0x0, 0x1, 0x1, 0x0, 0x2abe3e0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1050 +0x37f
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed.func1(0x7fb65ac918b8, 0xc424731f80, 0xc422be8f20, 0xc423f83d28, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:840 +0x270
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec(0xc422be8f20, 0x7fb65ac918b8, 0xc424731f80, 0x101, 0xc4217173b0, 0x1508908d, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:722 +0xa8
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed(0xc420408480, 0xc422493000, 0xc424731f00, 0x1, 0x1, 0x0, 0x0, 0xc42001e000, 0x604b80)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:861 +0x421
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execRequest(0xc420408480, 0xc422493000, 0xc421aeb4b3, 0x16, 0x0, 0x0, 0x16, 0xb4d)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:748 +0x489
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).ExecuteStatements(0xc420408480, 0xc422493000, 0xc421aeb4b3, 0x16, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:629 +0x129
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).handleSimpleQuery(0xc421e3d200, 0xc421e3d228, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:612 +0x148
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).serve(0xc421e3d200, 0x7fb65ac915f8, 0xc420c3aba0, 0xc4211847c0, 0x5400, 0xc4200a1768, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:503 +0xbd0
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).ServeConn(0xc4200a1600, 0x7fb65ac918b8, 0xc421a04b40, 0x2ac49e0, 0xc421081220, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:421 +0x96c
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func9.1(0x2ac49e0, 0xc421081220)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:746 +0x149
github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith.func1(0xc4200915f0, 0x7fb65ac915f8, 0xc420124fc0, 0xc42000e2e0, 0x2ac49e0, 0xc421081220, 0xc42162b6b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:140 +0xa9
created by github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith
	/go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:142 +0x267
```

Full logs at https://drive.google.com/a/cockroachlabs.com/file/d/0B7dLrdLwm9g5WmttcmpxUTZtc0k/view?usp=sharing for CRL employees (not sharing publicly due to insecure IP addresses)

--- a/pkg/sql/crdb_internal.go
+++ b/pkg/sql/crdb_internal.go
@@ -818,6 +818,7 @@ func populateSessionsTable(
 				parser.DNull,
 				parser.DNull,
 				parser.DNull,
+				parser.DNull,
 			); err != nil {
 				return err
 			}

